packages/contracts/src/dollar/mocks/MockBondingShareV2.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Pausable.sol";
+import "../core/UbiquityDollarManager.sol";
+import "../utils/SafeAddArray.sol";
+
+contract BondingShareV2 is ERC1155, ERC1155Burnable, ERC1155Pausable {
+    using SafeAddArray for uint256[];
+    struct Bond {
+        // address of the minter
+        address minter;
+        // lp amount deposited by the user
+        uint256 lpFirstDeposited;
+        uint256 creationBlock;
+        // lp that were already there when created
+        uint256 lpRewardDebt;
+        uint256 endBlock;
+        // lp remaining for a user
+        uint256 lpAmount;
+    }
+
+    UbiquityDollarManager public manager;
+    // Mapping from account to operator approvals
+    mapping(address => uint256[]) private _holderBalances;
+    mapping(uint256 => Bond) private _bonds;
```

Comment:
Bonds I think were changed to StakingToken but you should refer to the other code to see.

Reply:
This is a mock contract to interact with the contract as it's currently deployed on mainnet.